src/library-authoring/author-library/LibraryAuthoringPage.jsx

```diff
@@ -434,6 +485,73 @@ const LibraryAuthoringPageHeader = connect(
   },
 )(injectIntl(LibraryAuthoringPageHeaderBase));
 
+const ContentTagsDrawer = ({ openContentTagsDrawer, setOpenContentTagsDrawer }) => {
+  const iFrameRef = useRef();
+
+  if (openContentTagsDrawer) {
+    document.body.classList.add('drawer-open');
+  } else {
+    document.body.classList.remove('drawer-open');
+  }
+
+  useEffect(() => {
+    const handleCloseMessage = (event) => {
+      if (event.data === 'closeManageTagsDrawer') {
+        setOpenContentTagsDrawer('');
+      }
+    };
+
+    const handleCloseEsc = (event) => {
+      if (event.key === 'Escape' || event.keyCode === 27) {
+        setOpenContentTagsDrawer('');
+      }
+    };
+
+    // Add event listener to close drawer when close button is clicked or ESC pressed
+    // from within the Iframe
+    window.addEventListener('message', handleCloseMessage);
+    // Add event listern to close the drawer when ESC pressed and focus outside iframe
+    // If ESC is pressed while the Iframe is in focus, it will send the close message
+    // to the parent window and it will be handled with the above event listener
+    window.addEventListener('keyup', handleCloseEsc);
```

Comment:
```suggestion
    // Add event listener to close the drawer when ESC pressed and focus outside iframe
    // If ESC is pressed while the Iframe is in focus, it will send the close message
    // to the parent window and it will be handled with the above event listener
    window.addEventListener('keyup', handleCloseEsc);
```

Question: Clicking outside of iframe and pressing Esc doesn't close it. Is this expected behaviour?

Reply:
@navinkarkera I wasn't able to reproduce this. I clicked outside the iframe (on the darker side of the page) and pressed Esc and and the drawer closed. I tried this on both chrome and firefox, is there something I'm missing in the reproducing steps?

Reply:
@yusuf-musleh Ohh, it does work. Sorry, one my extensions (Tridactyl) was interfering with esc button.